src/cmd/initContainer.go

```diff
@@ -407,10 +408,60 @@ func mountBind(containerPath, source, flags string) error {
 	return nil
 }
 
+// redirectPath serves for creating symlinks for mainly crucial system files to
+// their counterparts on the host's filesystem.
+//
+// containerPath and target must be absolute paths
+//
+// If the target is a symlink, redirectPath will evaluate the link. If it is valid
+// than redirectPath will link containerPath to target. If it is not, then
+// redirectPath will still proceed with the linking in two different ways depending
+// on target being an absolute or a relative link:
+//
+// - absolute - target will be prepended with /run/host (link may not be invalid)
+//
+// - relative - target will not be changed (the resulting link will be invalid)
+//
+// folder signifies the target is a folder
+//
+// Example: containerPath /etc/resolv.conf and target /run/host/etc/resolv.conf with
+// systemd-resolved enabled on the host will create a link to /run/host/etc/resolv.conf
+// that links to /run/host/run/systemd/resolved/resolv-stub.conf
 func redirectPath(containerPath, target string, folder bool) error {
+	var err error
+	var resolvedTarget string
+
+	if !filepath.IsAbs(containerPath) || !filepath.IsAbs(target) {
+		panic("argument needs to be an absolute path")
+	}
+
+	logrus.Debugf("Preparing to redirect %s to %s", containerPath, target)
+
+	// Skip any filepath resolution if target is not a symlink
+	fi, err := os.Lstat(target)
+	if err != nil {
+		return fmt.Errorf("failed to lstat %s: %w", target, err)
+	}
+
+	if (fi.Mode() & os.ModeSymlink) != 0 {
```

Comment:
Looking at these deeply nested `if` branches, I am tempted to think that it would be better to split things out into a separate helper function. Something like `sanitizeTarget` or whatever.

Deeply nested structures are hard to read and follow; and we have quite a few subtle conditions based on whether we have an error or not, and from which operation.